flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/stream/StreamPhysicalLookupJoinRule.scala

```diff
@@ -76,8 +77,15 @@ object StreamPhysicalLookupJoinRule {
     val cluster = join.getCluster
 
     val providedTrait = join.getTraitSet.replace(FlinkConventions.STREAM_PHYSICAL)
-    val requiredTrait = input.getTraitSet.replace(FlinkConventions.STREAM_PHYSICAL)
-
+    var requiredTrait = input.getTraitSet.replace(FlinkConventions.STREAM_PHYSICAL)
+    val partitionJoinHint = join.getHints
+      .stream()
+      .filter(hint => JoinStrategy.isShuffleHashHint(hint.hintName))
+      .findFirst()
+    // if partitioning enabled, use the join key as partition key
+    if (partitionJoinHint.isPresent && !joinInfo.pairs().isEmpty) {
```

Comment:
I think there is something wrong with the way to determine whether SHUFFLE_HASH is enabled. If I mistakenly use SHUFFLE_HASH('left_table'), SHUFFLE_HASH will still be enabled instead of ignoring the hint. In fact, you should at least check whether the table in hint is the right table. And this should be reflected in the test.